### PR TITLE
Misc. prod readiness & presence validation on /login-options email-address page

### PR DESCRIPTION
**Pull request by embarnard**

### CODE REVIEW

File: app/controllers/portal/client_logins_controller.rb [hunk 1]
```diff
@@ -40,7 +40,7 @@ def check_verification_code
         hashed_verification_code = VerificationCodeService.hash_verification_code_with_contact_info(params[:contact_info], params[:verification_code])
         @records = client_login_service.login_records_for_token(hashed_verification_code)
         return if redirect_locked_clients # check if any records are already locked
-        if @records.present? # we have at least one match and none are locked
+        if @records.present? #|| (Rails.configuration.allow_magic_verification_code && params[:verification_code] == "000000") # we have at least one match and none are locked
```
Comment (embarnard) [inline app/controllers/portal/client_logins_controller.rb:43]:
```suggestion
        if @records.present? # we have at least one match and none are locked
```

File: app/controllers/state_file/intake_logins_controller.rb [hunk 2]
```diff
@@ -32,8 +37,14 @@ def update
       end
     end
 
+    def account_locked; end
```
Comment (embarnard) [inline app/controllers/state_file/intake_logins_controller.rb:40]: Adds this page
Screenshot 2024-01-25 at 12 30 29 PM

File: app/controllers/state_file/intake_logins_controller.rb [hunk 3]
```diff
@@ -32,8 +37,14 @@ def update
...
+
     private
 
+    def redirect_locked_clients
```
Comment (embarnard) [inline app/controllers/state_file/intake_logins_controller.rb:44]: commenting out this redirect for now b/c running into bug, will resolve in separate PR

File: app/views/state_file/questions/email_address/edit.html.erb [hunk 4]
```diff
@@ -4,9 +4,11 @@
   <h1 class="h2" id="main-question"><%= title %></h1>
 
   <%= form_with model: @form, url: { action: :update }, local: true, method: "put", builder: VitaMinFormBuilder do |f| %>
+    <div class="blue-group">
```
Comment (embarnard) [inline app/views/state_file/questions/email_address/edit.html.erb:7]: Formats these pages to look like figma
Screenshot 2024-01-25 at 12 31 58 PM

File: app/forms/state_file/email_address_form.rb [hunk 5]
```diff
@@ -3,6 +3,7 @@ class EmailAddressForm < QuestionsForm
     set_attributes_for :intake, :email_address
 
     validates :email_address, 'valid_email_2/email': true
+    validates :email_address, presence: true
```
Comment (embarnard) [inline app/forms/state_file/email_address_form.rb:6]: currently email address page on the login-options flow allows you to proceed without submitting anything

File: app/controllers/state_file/state_file_pages_controller.rb [hunk 6]
```diff
@@ -8,6 +8,8 @@ def redirect_locale_home
     end
 
     def fake_direct_file_transfer_page
+      # TODO: make sure we can't reach this page on prod
```
Comment (mpidcock) [inline app/controllers/state_file/state_file_pages_controller.rb:11]: Should we add an environment check? What needs to be done here?

File: app/views/state_file/questions/initiate_data_transfer/edit.html.erb [hunk 7]
```diff
@@ -4,15 +4,16 @@
...
+    <%= link_to @link.to_s, class: "button button--primary button--icon", id: "firstCta" do %>
+      <%= image_tag("icons/link-white.svg", alt: "") %>
+      <%= t(".button") %>
+    <% end %>
 
-  <% if ENV["IRS_TRANSFER_AUTH_URL"].present? %>
-    <%= link_to @irs_testing_link.to_s, class: "button", id: "firstCta" do %>
-      Transfer Data from IRS testing site (Don't click this unless you've been asked to do so)
+    <% if ENV["IRS_TRANSFER_AUTH_URL"].present? %>
+      <%= link_to @irs_testing_link.to_s, class: "button button--full-width text--small", id: "firstCta" do %>
```
Comment (mpidcock) [inline app/views/state_file/questions/initiate_data_transfer/edit.html.erb:14]: Is this what you were bringing up in our group discussion earlier? It sounds like we want this button to be fixed up to be nicer and work for either testing or production depending on the environment's credentials.

File: app/views/state_file/questions/phone_number/edit.html.erb [hunk 8]
```diff
@@ -3,10 +3,13 @@
 <% content_for :card do %>
   <h1 class="h2" id="main-question"><%= title %></h1>
 
-  <%= form_with model: @form, url: { action: :update }, local: true, method: "put", builder: VitaMinFormBuilder do |f| %>
+    <%= form_with model: @form, url: { action: :update }, local: true, method: "put", builder: VitaMinFormBuilder do |f| %>
+    <div class="blue-group">
     <%= f.cfa_input_field(:phone_number, t(".phone_number_label"), classes: ["form-width--long"]) %>
+    </div>
```
Comment (mpidcock) [inline app/views/state_file/questions/phone_number/edit.html.erb:9]: Nit: I think the indentation is off here

**Review by mpidcock — COMMENTED** [review]
Adding TODOs makes me nervous so I'd prefer if those could be addressed before merging

File: app/controllers/state_file/state_file_pages_controller.rb (hunk 6 above)
Comment (embarnard) [reply app/controllers/state_file/state_file_pages_controller.rb:11]: @mpidcock yeah actually we did changes here instead so going to remove all the link stuff from this PR https://www.pivotaltracker.com/story/show/186917213

File: app/views/state_file/questions/initiate_data_transfer/edit.html.erb (hunk 7 above)
Comment (embarnard) [reply app/views/state_file/questions/initiate_data_transfer/edit.html.erb:14]: @mpidcock yes going to remove it and leave styling changes for here https://www.pivotaltracker.com/story/show/186918612

**Review by mpidcock — APPROVED** [review]

**Pull request merged**